File: src/main/java/io/github/chase22/infrastructure/server/endpoint/EndpointRegistry.java
package io.github.chase22.infrastructure.server.endpoint;

import io.github.chase22.api.endpoints.GetMeEndpoint;
import io.github.chase22.infrastructure.server.TelegramApiEndpoint;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.HashMap;
import java.util.Map;

public class EndpointRegistry {
    private final Logger LOGGER = LoggerFactory.getLogger(EndpointRegistry.class);

    private Map<String, TelegramApiEndpoint> endpointMap = new HashMap<>();

    public EndpointRegistry() {
        registerTelegramEndpoints();
    }

    public TelegramApiEndpoint getForPath(String path) {
        return endpointMap.get(path);
    }

    private void registerTelegramEndpoints() {
        addEndpoint(new GetMeEndpoint());
    }

    private void addEndpoint(TelegramApiEndpoint servlet) {
        LOGGER.info("Registering servlet " + servlet.getClass().getName() + " with mapping " + servlet.getPath());
        endpointMap.put(servlet.getPath(), servlet);
    }
}

File: src/main/kotlin/io/github/chase22/infrastructure/server/endpoint/EndpointRegistry.kt
package io.github.chase22.infrastructure.server.endpoint

import com.fasterxml.jackson.databind.ObjectMapper
import io.github.chase22.api.endpoints.GetMeEndpoint
import io.github.chase22.infrastructure.server.TelegramApiEndpoint
import io.github.chase22.telegram.telegrambotapimock.api.data.User
import org.slf4j.LoggerFactory

import java.util.HashMap

class EndpointRegistry(objectMapper: ObjectMapper, bot: User) {

    private val endpointMap = HashMap<String, TelegramApiEndpoint<*>>()

    init {
        addEndpoint(GetMeEndpoint(bot, objectMapper))
    }

    fun getForPath(path: String): TelegramApiEndpoint<*>? = endpointMap[path]

    private fun addEndpoint(servlet: TelegramApiEndpoint<*>) {
        LOGGER.info("Registering servlet " + servlet.javaClass.name + " with mapping " + servlet.path)
        endpointMap[servlet.path] = servlet
    }

    companion object {
        private val LOGGER = LoggerFactory.getLogger(EndpointRegistry::class.java)
    }
}
